### PR TITLE
doc: add vite-plugin-node

### PR DESCRIPTION
**Pull request by axe-me**
### Checklist

- [x] Title as described.
- [x] Make sure you put things in the right category.
- [x] The description of your item should be a sentence with less than 24 words.
- [x] Omit unnecessary words already provided in the context (e.g. `for Vite`, `a Vite plugin`).
- [x] Always add your items to the end of a list.

### Plugins/Tools

- [x] The plugin/tool is working with **Vite 2.0 and onward**.
- [x] The project is Open Source.
- [ ] The repo should be at least 30 days old.
- [x] The documentation is in English.
- [x] The project is active and maintained.
- [x] The project accepts contributions.
- [x] Not a commercial product

### Apps/Websites

- [ ] The website is available without errors and load in a reasonable amount of time.
- [ ] The website is using Vite intensively.
- [ ] The website is original and not too simple. For that reason, blogs and simple landing pages are rejected.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,6 +143,10 @@ A curated list of awesome things related to <a href='https://github.com/vitejs/v
 - [vite-plugin-mocker](https://github.com/minjs1cn/vite-plugin-mocker) - Mocker server.
 - [vite-plugin-test](https://github.com/aelbore/vite-plugin-test) - Headless testing your component.
 
+#### Dev Servers
+
+- [vite-plugin-node](https://github.com/axe-me/vite-plugin-node) - Use Vite as node dev server.
```
Comment (antfu) [inline README.md:148]: The plugin looks good to me. But it took me a while to understand what's it trying to solve. Can you update the description to be more clear and accurate? Thanks.

**Review by antfu — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (axe-me) [reply README.md:148]: wording is hard for programmer lol

Comment (antfu) [reply README.md:148]: Integration with Node.js backend servers?

Comment (axe-me) [reply README.md]: thx, updated.

**Comment by axe-me** [comment]
btw I noticed there are two `Bundling` in Table of content when I resolve merge conflict.

File: README.md [hunk 2]
```diff
@@ -156,6 +157,10 @@ A curated list of awesome things related to <a href='https://github.com/vitejs/v
 - [vite-plugin-importer](https://github.com/ajuner/vite-plugin-importer) - Integration for babel-plugin-import.
 - [vite-plugin-banner](https://github.com/chengpeiquan/vite-plugin-banner) - Adds a banner to the top of each generated chunk.
 
+#### Dev Servers
+
+- [vite-plugin-node](https://github.com/axe-me/vite-plugin-node) - Integration with Node.js backend servers.
```
Comment (antfu) [inline README.md:162]: Let's move it to the #Integrations section for now. We will see if need to separate this out when we have enough plugins in this category.

Comment (axe-me) [reply README.md]: moved, also fixed the duplicated bundling sections

**Pull request merged**